util/httputil/port_pool.go

```diff
@@ -0,0 +1,66 @@
+package httputil
+
+import (
+	"fmt"
+	"sync"
+)
+
+// [1500] => true // busy
+// [1501] => true // busy
+// [1502] => false // free
+// last port assigned => 1501
+var portRangeInit = 14100
+var portRangeSize = 100
```

Comment:
I have two comments.
One is that this info should be confiburable, as user may want to tweak, 99.9% the defaults should be ok, but better to have it available as we have with other connection parameters.

Another idea is we need to think in the concurrency scenarios, in two ways, when a single user access to a high populated cluster and when there are multiple users accesing Kiali.

Probably nothing to change that just to calculate in which certain situations this may fail, i.e. we fix some concurrency issue in a very big cluster for a similar case.

Reply:
I think that using the portforward is mostly on fetching proxy status, control plane status, and fetching envoy details.
So, per request it should be a few, and then the limit would be in the number of users that may access to Kiali, but just wanted to check if I missed some case (i.e. when fetching a list of workloads) that may impact when number is > 100.

Reply:
I have pros and cos of adding the pool params into settings. 
Will other processes in the kiali pod open ports in the same port space? Won't Kiali serve will always be the only process opening ports in the kiali pod?

Reply:
And agree with the second message. The port range size limits the number of users hitting the same button at the same time. However, it is pretty unlikely to reach the number of 100. Let's list the number of calls done and how they are cache:

1) Istio Component status - One port open per each IstioD pod (1-N). This request is cached and it is regenerated every X seconds. (Extreme scenario: 100 users send the same request at the same precise moment and the server processes the request at the same moment and the cache is not generated/expired).
2) Service Registry status - One port for each IstioD pod (1-N). Same as before. Content cached. There should be one request after the cache is expired.
3) Proxy Status - One port for each IstioD pod (1-N). Same as before. Content is cached. There should be one request after the cache is expired.
4) Envoy Config endpoints: One port per user click in the envoy modal. Therefore, if kiali is used with high concurrency, this might reach the limit soon. We might consider applying a short cache for this. However, this cache could be huge because is linearly depending on how many workloads you have in the mesh.


We could improve this pool mechanism for high saturated networks where the calls to the sidecars or Istiods takes a lot of time. User starts to send requests. Those requests are stacked/queued because of waiting for istiod/sidecar to respond. The pool port is empty. Then Kiali starts erroring that there are no available ports.

I don't think there are istio meshes with 25 istiod pods managing the mesh. This would mean that only there is the possibility of 25 users clicking the envoy tab at the same time. The number 26th would get an "port is already is use".
100 ports = 25 ports for Istio Component status + 25 ports for Service Registy + 25 ports for Proxy Status + 25 concurrent users clicking the envoy tab.

Reply:
Is there a reason you use 100 there? Ports are cheap, no? Use 1,000 and forget about it :)

Reply:
I support @jmazzitelli comment :) it's safe to put a gib number in the ports as soon as we control that range and probably we can revisit this in the context of high scalability scenarios (where we are not yet there).